### PR TITLE
Improve handling of resolved vs unresolved checker.

### LINKED ISSUE #179: fix "add signal to architecture" uses undefined instead of std_(u)logic
image

### PR DESCRIPTION
**Pull request by g0t00**
No hard-coded types anymore. Checking if a type is unresolved/resolved and comparing to settings.
If different try to find subtype with correct resolution status.
Also split the setting for signal/port.
Rule is now also working for signals. I think that was broken before.
Added a bunch of automated tests.

Fixed #179
should obsolete #147

### CODE REVIEW

File: lib/rules/type-resolved.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+      if (definition instanceof OSubType && definition.resolved) {
+        resolved = true;
+      }
+      // Type referenced is not resolved. Now try to find resolved subtype of this type or aliases of it
+      if (!resolved) {
+        const root = definition.getRootElement();
+        let resolvedSubtype: string[] = [];
+        for (const subtype of root.types) {
+          if (subtype instanceof OSubType && subtype.resolved && alias.indexOf(subtype.superType.referenceToken.getLText()) > -1)
+            resolvedSubtype.push(subtype.lexerToken.text);
```
[the remark this answers is not in the document]
Comment (Nik-Sch) [reply lib/rules/type-resolved.ts:84]: is there a linter rule for this?

File: lib/rules/type-resolved.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                    , unresolvedType)]
+                }
+              },
+              CodeActionKind.QuickFix)));
+            return actions;
+          });
+          this.addMessage({
+            range: object.typeReference[0].range,
+            severity: DiagnosticSeverity.Information,
+            message: `Port is using unresolved type (${object.typeReference.map(a => a.referenceToken.text).join()}) should use resolved subtype ${resolvedSubtype.join(' or ')} `,
```
Comment (Nik-Sch) [inline lib/rules/type-resolved.ts:111]:
```suggestion
            message: `Port is using unresolved type (${object.typeReference[0].referenceToken.text}) should use resolved subtype ${resolvedSubtype.join(' or ')} `,
```

File: lib/rules/type-resolved.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                  , unresolvedType)]
+              }
+            },
+            CodeActionKind.QuickFix)));
+          return actions;
+        });
+        this.addMessage({
+          range: object.typeReference[0].range,
+          severity: DiagnosticSeverity.Information,
+          message: `Port is using resolved subtype (${object.typeReference.map(a => a.referenceToken.text).join()}) should use unresolved type ${unresolvedTypes.join(' or ')} `,
```
Comment (Nik-Sch) [inline lib/rules/type-resolved.ts:48]:
```suggestion
          message: `Port is using resolved subtype (${object.typeReference[0].referenceToken.text}) should use unresolved type ${unresolvedTypes.join(' or ')} `,
```

File: lib/rules/type-resolved.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (g0t00) [reply lib/rules/type-resolved.ts:84]: Yes, added.

**Pull request merged**